### PR TITLE
[5.0] CMSApplication: Switching to triggerEvent

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This syncs the event triggers in `CMSApplication::execute()` to use `triggerEvent()` overall. I know that `triggerEvent()` is deprecated and that in the end we want to switch to event classes instead. However right now this is not implemented everywhere and the task for someone else. But: The way the event is triggered right now throws a deprecation warning and we are using `triggerEvent()` in the rest of the method, so I'd rather keep it consistent for now.

### Testing Instructions
Basically a codreview... Make sure that the events are still triggered, I guess?

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Application/CMSApplication.php [hunk 1]
```diff
@@ -324,13 +324,13 @@ public function execute()
         }
 
         // Trigger the onBeforeRespond event.
-        $this->getDispatcher()->dispatch('onBeforeRespond');
+        $this->triggerEvent('onBeforeRespond');
```
Comment (joomdonation) [inline libraries/src/Application/CMSApplication.php]: Instead of using deprecated `triggerEvent` method, how about changing this to `$this->dispatchEvent('onBeforeRespond');` , see https://github.com/joomla-framework/application/blob/2.0-dev/src/AbstractApplication.php#L91 for details implementation.

Comment (Hackwar) [reply libraries/src/Application/CMSApplication.php]: In this PR I'm only going to unify this to use the same code for events in the whole method. If you want to change this to other code, please open a new PR.

**Comment by Hackwar** [comment]
Since I don't see this being merged into 4.2-dev, I've changed the base branch to 4.3-dev. Lets hope it will be merged in there.

File: libraries/src/Application/CMSApplication.php (hunk 1 above)
Comment (Hackwar) [reply libraries/src/Application/CMSApplication.php]: I want to get this solved, so I changed all in this method to $this->dispatchEvent().

**Comment by joomdonation** [comment]
I tested the changes with different kind of listeners:

```php
public function onBeforeRespond()
{
}
```

```php
public function onBeforeRespond(\Joomla\Event\Event $event)
{
}
```

```php
public function onBeforeRespond(\Joomla\Application\Event\ApplicationEvent $event)
{
}
```

The first two are for the old way to trigger event (using `triggerEvent` method), the last one is for new way of event trigger (using `dispatchEvent` method). All works good, so this change look fine to me. Could anyone else take a look at this PR, too ? Maybe @Fedik ?

Another question is if this change is OK, should we do the same for `SiteApplication` and `AdministratorApplication` ?

**Comment by Hackwar** [comment]
Since this change will most likely not go into 4.3, I've rebased it to 4.4.

**Comment by Fedik** [comment]
That looks and works good to me. What about Site, Admin and other applications?

**Comment by laoneo** [comment]
Please rebase to 5.0, safer to use the new way in the next major. Thanks for understanding.

**Comment by Fedik** [comment]
This was fixed as part of #40522 Implementing application event classes.
I close this PR. Thanks for your work :wink:
